WowUp.WPF/Services/AddonService.cs

```diff
@@ -340,11 +343,26 @@ public async Task UninstallAddon(Addon addon)
                 await FileUtilities.DeleteDirectory(addonDirectory);
             }
 
+            await UninstallDependencies(addon);
+
             _addonRepository.DeleteItem(addon);
 
             AddonUninstalled?.Invoke(this, new AddonEventArgs(addon, AddonChangeType.Uninstalled));
         }
 
+        public async Task UninstallDependencies(Addon addon)
+        {
+            var addonDependencies = _dependencyRepository.GetAddonDependencies(addon);
+            foreach (var dependency in addonDependencies)
+            {
+                var dependencyAddon = GetAddon(dependency.DependencyId);
+                if (dependencyAddon != null && _dependencyRepository.GetDependentAddons(dependencyAddon).All(dep => dep.AddonId == addon.Id)) 
+                    await UninstallAddon(dependencyAddon);
+                _dependencyRepository.DeleteItem(dependency);
+            }
+            _dependencyRepository.DeleteItems(_dependencyRepository.GetDependentAddons(addon));
```

Comment:
Is this redundant or am i just sleep deprived?

Reply:
this is actually needed if a dependency of an addon is deleted first and only, otherwise there are abondended entries in the database, that will only resolve after the main addon is deleted.